includes/Services/FlowService.php

```diff
@@ -0,0 +1,254 @@
+			if ( self::is_array_indexed( $params[ $key ] ) ) {
+				// If the Database value is empty or an Indexed Array, to avoid Associative arrays to be overwritten (Eg: data)
+				( ( count( $value ) === 0 ) || ( count( array_filter( array_keys( $value ), 'is_string' ) ) === 0 ) ) ?
+					$flow_data[ $key ]   = $params[ $key ]
+					: $flow_data[ $key ] = $value;
+				continue;
+			}
+
+			// To handle Associative Arrays gracefully
+			$nested_flow_data = self::update_post_call_data_recursive( $value, $params[ $key ] );
+			if ( \is_wp_error( $nested_flow_data ) ) {
+				return $nested_flow_data;
+			}
+			$flow_data[ $key ] = $nested_flow_data;
+		}
+		return $flow_data;
+	}
+
+	/**
+	 * Logic to check for an Indexed Array
+	 *
+	 * @param array $array To verify for an Indexed Array
+	 *
+	 * @return boolean
+	 */
+	private static function is_array_indexed( $array ) {
+		return count( array_filter( array_keys( $array ), 'is_string' ) ) === 0;
+	}
+
+	/**
+	 * Read onboarding flow options from database
+	 *
+	 * @return array|false
+	 */
+	public static function read_flow_data_from_wp_option() {
+		return \get_option( Options::get_option_name( 'flow' ), false );
+	}
+
+	/**
+	 * Check the current flow type and update default data if flowtype is ecommerce.
+	 *
+	 * @param array $data default flow data.
+	 *
+	 * @return array
+	 */
+	private static function update_default_data_for_ecommerce( $data ) {
+		// update default data with ecommerce data
+		$data['data']['topPriority']['priority1'] = 'selling';
+		$data['data']['siteType']['referTo']      = 'business';
+		return $data;
+	}
+
+	/**
+	 * Function to search for key in array recursively with case sensitive exact match
+	 *
+	 * @param array  $params Param Data
+	 * @param array  $flow_data Options Data
+	 * @param string $header_key Array Level in Recursion
+	 *
+	 * @return string
+	 */
+	public static function find_mismatch_key( $params, $flow_data, $header_key = 'Base Level' ) {
+		$exception_list = Flows::get_exception_list();
+		foreach ( $params as $key => $value ) {
+			if ( ! isset( $exception_list[ $key ] ) && is_array( $flow_data ) ) {
+				// Error if the key added by the user is not present in the database
+				if ( ! array_key_exists( $key, $flow_data ) ) {
+					return new \WP_Error(
+						'wrong_param_provided',
+						'Wrong Parameter Provided',
+						array(
+							'status'                  => 400,
+							'Mismatched Parameter(s)' => $header_key . ' => ' . $key,
+						)
+					);
+				}
+
+				// To check sub-Arrays
+				if ( ! is_array( $value ) || empty( $value ) ) {
+					continue;
+				}
+
+				// For Indexed Arrays
+				if ( count( array_filter( array_keys( $value ), 'is_string' ) ) === 0 ) {
+					foreach ( $value as $index_key => $index_value ) {
+						// For Indexed Arrays having Associative Arrays as Values
+						if ( is_array( $index_value ) ) {
+							self::find_mismatch_key( $value[ $index_key ], $flow_data[ $key ][ $index_key ], $key . ' : ' . $index_key );
```

Comment:
Are we missing a `return` here? Cause we have called the function but it does not do anything? I'm thinking we may not have caught this cause all these cases are in the exception list and are skipped?